pkg/shell/superuser.jsx

```diff
@@ -203,10 +241,17 @@ const LockDialog = ({ proxy, host }) => {
     const [error, setError] = useState(null);
 
     const apply = () => {
+        const was_sudo = proxy.Current == "sudo";
         setError(null);
         proxy.Stop()
                 .then(() => {
-                    return cockpit.spawn(["sudo", "-k"], { host: host }).always(() => {
+                    function maybe_reset_sudo() {
+                        if (was_sudo)
+                            return cockpit.spawn(["sudo", "-k"], { host: host });
+                        else
+                            return Promise.resolve();
+                    }
```

Comment:
Indeed this is awkward .. I like your proposal of adding a "stop" command, to generalize this. That can/should be a follow-up, though.

However, why doesn't this just kill the root bridge -- that should be enough to drop administrative privs, regardless of the mechanism? But I think I don't fully understand this -- `sudo -k` would not kill the existing root bridge anyway, so I suppose that happens someplace else.

Reply:
The priv bridge is indeed stopped correctly and admin privs are lost because of that stopping, regardless of whether `sudo -k` is run or not.  We run `sudo -k` to remove any cached credentials, so that the user will be prompted again for their password on the next attempt to gain admin privs.  Otherwise the usual sudo timeout would apply, and the user would get admin privs immediately without password if they try to get them back fast enough after dropping them.

I find the current behavior more consistent and easier to predict.  The credential caching makes sense when using sudo on the command line for exevery individual command, but not when switching the whole UI permenantly into another mode.

Reply:
Heh, we can use `-k` when starting the priv bridge... I try that.

Reply:
I see, so that happens in addition to killing the bridge. Indeed it would then make more sense to just call `sudo -k cockpit-bridge` right away during the start command, to entirely bypass the caching.

Reply:
Works!